app/server/utils/context_processors.py

```diff
@@ -1,36 +1,37 @@
 from django.utils import timezone
 from django.http import HttpRequest
-from server.utils.constants import DEFAULT_TIME_FORMAT
-from server.utils.time_manager import format_in_zone, now
-from server.users.models import User
+from server.utils.time_manager import format_in_zone
 from typing import TypedDict
 
-class UserContext(TypedDict):
-    is_authenticated: bool = False
-    is_staff: bool = False
-    email: str = ""
-    first_name: str = ""
-    last_name: str = ""
 
-def user_context(request: HttpRequest) -> UserContext:
-    context = UserContext()
+class UserInfo(TypedDict):
+    is_authenticated: bool
+    is_staff: bool
+    email: str
+    first_name: str
+    last_name: str
+
 
-    if hasattr(request, "user"):
-        user = request.user
-        if (not user.is_anonymous):
-            context["is_authenticated"] = user.is_authenticated
-            context["is_staff"] = user.is_staff
-            context["email"] = user.email
-            context["first_name"] = user.first_name
-            context["last_name"] = user.last_name
+class UserContext(TypedDict):
+    user: UserInfo
 
 
-    return context
+def user_context(request: HttpRequest) -> UserContext:
+    return {
+        "user": {
+            "is_authenticated": request.user.is_authenticated,
+            "is_staff": request.user.is_staff,
+            "email": request.user.email if not request.user.is_anonymous else "",
```

Comment:
I think `request.user.is_anonymous or ""` will suffice.

Applies to `request.user.first_name` and request.user.last_name` as well.

Reply:
I don't really understand. 

If you mean `"email": request.user.is_anonymous or ""`, then it doesn't make sense, because we need to provide the `request.user.email` value

If you mean `"email": request.user.email or ""`, then it also won't work, because if user is anonymous the `user.email` field doesn't exist (since then user is of type `AnonymousUser`, not `User`), so we get an exception.